lib/mix/phoenix.ex

```diff
@@ -3,13 +3,24 @@ defmodule Mix.Phoenix do
   @moduledoc false
 
   @doc """
-  Returns the module base name based on the application name.
+  Returns the module base name based on the configuration value.
+
+      config :my_app
+        phoenix_namespace: My.App
+
   """
   def base do
-    Mix.Project.config
-    |> Keyword.fetch!(:app)
-    |> to_string
-    |> Phoenix.Naming.camelize
+    app = Mix.Project.config |> Keyword.fetch!(:app)
+
+    Mix.Config.read!("config/config.exs")
```

Comment:
Oh, we don't need to read the config. When we got read, we could assume the project was already configured. So all we need to do is invoke `Application.get_env(app, :phoenix_namespace)`. :)